File: AILang/benchmarks/python/fib_mix.py
def fib_mix_bench(iterations: int) -> int:
    a = 0
    b = 7
    i = 0
    while i < iterations:
        a = a + b
        if a > 1_000_000_000:
            a = a - 1_000_000_000
        b += 1
        if b > 1_000_000_000:
            b = b - 1_000_000_000
        i += 1
    return a


def main() -> int:
    iterations = 8000000
    result = fib_mix_bench(iterations)
    print(result)
    return 0


if __name__ == "__main__":
    raise SystemExit(main())

File: AILang/benchmarks/rust/fib_mix.rs
fn fib_mix_bench(iterations: u32) -> u32 {
    let mut a: u32 = 0;
    let mut b: u32 = 7;
    let mut i: u32 = 0;
    while i < iterations {
        a = a + b;
        if a > 1_000_000_000 {
            a -= 1_000_000_000;
        }
        b += 1;
        if b > 1_000_000_000 {
            b -= 1_000_000_000;
        }
        i += 1;
    }
    a
}

fn main() {
    let iterations: u32 = 8000000;
    let result = fib_mix_bench(iterations);
    println!("{}", result);
}
